src/api/VoteMonitor.Api.DataExport/Handlers/DataExportQueryHandler.cs

```diff
@@ -77,51 +57,62 @@ public async Task<List<ExportModel>> Handle(GetDataForExport request, Cancellati
 		WHERE
 			a.LastModified >= @from
             AND obs.IsTestObserver = 0
-
             ";
 
-            var parameters = new List<SqlParameter>
-            {
-                new SqlParameter("@from", request.From ?? new DateTime(2019, 11, 08, 6, 0, 0)),
-
-            };
+            DynamicParameters parameters = new DynamicParameters();
+            parameters.Add("from", request.From ?? new DateTime(2019, 11, 08, 6, 0, 0));
 
             if (request.ApplyFilters)
             {
                 if (request.To.HasValue)
                 {
                     query += " AND a.LastModified <= @to ";
-                    parameters.Add(new SqlParameter("@to", request.To ?? DateTime.Now.AddDays(2)));
+                    parameters.Add("to", request.To ?? DateTime.Now.AddDays(2));
                 }
 
                 if (request.ObserverId.HasValue)
                 {
                     query += " AND obs.Id = @ObserverId ";
-                    parameters.Add(new SqlParameter("@ObserverId", request.ObserverId));
+                    parameters.Add("ObserverId", request.ObserverId);
                 }
 
                 if (request.NgoId.HasValue)
                 {
                     query += " AND obs.IdNgo = @IdNgo ";
-                    parameters.Add(new SqlParameter("@IdNgo", request.NgoId));
+                    parameters.Add("IdNgo", request.NgoId);
                 }
 
                 if (!string.IsNullOrEmpty(request.County))
                 {
                     query += " AND a.CountyCode = @County ";
-                    parameters.Add(new SqlParameter("@County", request.County));
+                    parameters.Add("County", request.County);
                 }
 
                 if (request.PollingStationNumber.HasValue)
                 {
                     query += " AND a.PollingStationNumber = @PollingStationNumber ";
-                    parameters.Add(new SqlParameter("@PollingStationNumber", request.PollingStationNumber));
+                    parameters.Add("PollingStationNumber", request.PollingStationNumber);
                 }
             }
 
-            var exportData = _context.ExportModels.FromSqlRaw(query, parameters.ToArray());
-
-            return await exportData.ToListAsync(cancellationToken);
+            IEnumerable<ExportModel> data = Enumerable.Empty<ExportModel>();
+            using (var db = _context.Database.GetDbConnection())
+            {
+                try
+                {
+                    db.Open();
+                    data = db.Query<ExportModel>(sql: query.ToString(), param: parameters, commandTimeout: 60);
+                }
+                catch (Exception e)
+                {
+                    _logger.LogError(e, nameof(Handle));
+                }
+                finally
+                {
+                    db.Close();
```

Comment:
Disposing DbConnection will also close the database so this call is redundant. Is it ok that we're not throwing the exception forward and it will be caught here?

I think we should throw it and in that case, the whole try-catch will be redundant.

Reply:
should resolved on: https://github.com/code4romania/monitorizare-vot/pull/319/commits/a4e5dac3f1e45e22ed2cde5a456b38e4e9d00da0

However, I've added a try-catch on higher level because I'm afraid to return to client the whole uncaught exception: this could be a security risk because of the information returned by the exception. What do you think?

Reply:
It shouldn't be a problem as the exception is not displayed(or should not be displayed) to users when the environment is "production". The exception stack trace is returned only on dev and staging by default and if someone changed that, we should fix it on the entire project(but I don't think so). I'll check tomorrow if running on prod, any stack trace is returned.

Reply:
![image](https://user-images.githubusercontent.com/40202887/96642893-79a24380-132f-11eb-978a-6e4df2bdb4c4.png)
I checked and the exception is not displayed to the user on a prod environment(the ASP.NET code default behavior).